src/presentation/wallet/send/choose-fee.tsx

```diff
@@ -1,368 +1,217 @@
+        const currentSatsPerByte = feeLevelToSatsPerByte[feeLevel];
+        const tx: string = w.buildTx(
+          w.createTx(),
+          recipients,
+          greedyCoinSelector(),
+          changeAddressGetter,
+          true,
+          currentSatsPerByte
+        );
+        setUnsignedPendingTx(tx);
+        return;
       }
-    }
-  }, [
-    app.network.value,
-    changeAddressGetter,
-    feeCurrency,
-    feeLevel,
-    receipients,
-    satsPerByte,
-    supportedAssets,
-    supportedAssets.length,
-    transaction.amountInSatoshi,
-    transaction.asset,
-    transaction.changeAddress,
-    transaction.feeChangeAddress,
-    transaction.receipientAddress,
-    unspents,
-  ]);
 
-  // Fetch topup utxo from Taxi
-  useEffect(() => {
-    void (async (): Promise<void> => {
-      if (
-        feeCurrency &&
-        feeCurrency !== lbtcAssetByNetwork(app.network.value) &&
-        Object.keys(transaction.taxiTopup).length === 0
-      ) {
-        try {
-          const taxiTopup = await fetchTopupFromTaxi(taxiURL[app.network.value], feeCurrency);
-          dispatch(setTopup(taxiTopup));
-        } catch (error) {
-          console.error(error.message);
-          setErrorMessage(error.message);
+      if (taxiAssets.includes(feeCurrency)) {
+        // taxi
+        const taxiTopup = await fetchTopupFromTaxi(taxiURL[network], feeCurrency);
+        await dispatch(setTopup(taxiTopup));
+        if (taxiTopup.topup === undefined) {
+          throw new Error('Taxi topup is undefined');
         }
-      }
-    })();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [app.network.value, feeCurrency, transaction.taxiTopup]);
+        const taxiPayout: RecipientInterface = {
+          value: taxiTopup.topup?.assetAmount,
+          asset: taxiTopup.topup?.assetHash,
+          address: '',
+        };
 
-  // Fill Taxi tx
-  useEffect(() => {
-    if (
-      Object.keys(transaction.taxiTopup).length !== 0 &&
-      feeCurrency === usdtAssetHash(assets[app.network.value])
-    ) {
-      const taxiPayout = {
-        value: transaction.taxiTopup.topup?.assetAmount,
-        asset: transaction.taxiTopup.topup?.assetHash,
-        address: '',
-      } as RecipientInterface;
-      try {
         const tx: string = fillTaxiTx(
-          transaction.taxiTopup.topup?.partial as string,
-          unspents,
-          receipients,
+          taxiTopup.topup.partial,
+          Object.values(wallet.utxoMap),
+          recipients,
           taxiPayout,
           greedyCoinSelector(),
           changeAddressGetter
         );
+
         setUnsignedPendingTx(tx);
-      } catch (error) {
-        console.log(error);
-        setErrorMessage(error.message);
+        return;
       }
+    } catch (err) {
+      console.error(err);
+      setErrorMessage(`Error for ${assets[feeCurrency].ticker || 'unknow'} asset`);
+      setFeeCurrency(undefined);
```

Comment:
In case of error feeCurrency is changed to undefined, which breaks the logo. Would be nice if we can avoid that.

Reply:
[here](https://github.com/louisinger/marina/blob/dee4a8301ffd763a040733cb44a9f3e2b80cedbb/src/presentation/wallet/send/choose-fee.tsx#L204) the `|| ''` should return the default logo path if feeCurrency is undefined